data/scripts/network/cyclopedia_character.lua

```diff
@@ -24,6 +24,87 @@ local function sendBasicInfo(self, msg)
 	return true
 end
 
+local function getLevelPercent(player)
+	local currentLevelExp = Game.getExperienceForLevel(player:getLevel())
+	local nextLvlExp = Game.getExperienceForLevel(player:getLevel() + 1)
+	local playerExp = player:getExperience()
+	return (playerExp - currentLevelExp) * 100 / (nextLvlExp - currentLevelExp)
+end
+
+local clientSkillsId = {
+	[0] = CYCLOPEDIA_SKILL_FIST,
+	[1] = CYCLOPEDIA_SKILL_CLUB,
+	[2] = CYCLOPEDIA_SKILL_SWORD,
+	[3] = CYCLOPEDIA_SKILL_AXE,
+	[4] = CYCLOPEDIA_SKILL_DISTANCE,
+	[5] = CYCLOPEDIA_SKILL_SHIELDING,
+	[6] = CYCLOPEDIA_SKILL_FISHING
+}
+
+local function sendGeneralStats(self, msg)
+	msg:addU64(self:getExperience())
+	msg:addU16(self:getLevel())
+	msg:addByte(math.floor(getLevelPercent(self)))
+
+	msg:addU16(self:getClientExpDisplay())
+	msg:addU32(0) -- turnament exp (deprecated)
+	msg:addU16(0) -- low level bonus
+	msg:addU16(0) -- store exp bonus
+	msg:addU16(self:getClientStaminaBonusDisplay())
+	msg:addU16(0) -- exp boost remaining time
+	msg:addByte(0) -- enable exp boost store button
+
+	msg:addU16(self:getHealth())
+	msg:addU16(self:getMaxHealth())
+	msg:addU16(self:getMana())
+	msg:addU16(self:getMaxMana())
+	msg:addByte(self:getSoul())
+	msg:addU16(self:getStamina())
+
+	local foodTime = 0
+	local condition = self:getCondition(CONDITION_REGENERATION, CONDITIONID_DEFAULT)
+	if condition then
+		foodTime = condition:getTicks() /  1000
+	end
+
+	msg:addU16(foodTime)
+	msg:addU16(self:getOfflineTrainingTime() / 60 / 1000)
+	msg:addU16(self:getSpeed() / 2)
+	msg:addU16(self:getBaseSpeed() / 2)
+
+	local infiniteCapacity = self:hasFlag(PlayerFlag_HasInfiniteCapacity) and 1000000
+	msg:addU32(infiniteCapacity or self:getCapacity()) -- base + bonus capacity
+	msg:addU32(infiniteCapacity or self:getCapacity())
+	msg:addU32(infiniteCapacity or self:getFreeCapacity())
+
+	msg:addByte(8) -- ??
+	msg:addByte(1) -- ??
+
+	-- magic level
+	msg:addU16(self:getMagicLevel())
+	msg:addU16(self:getBaseMagicLevel())
+
+	-- loyalty bonus
+	msg:addU16(self:getBaseMagicLevel())
```

Comment:
Base magic level twice?

Reply:
it's the magic level information and includes the loyalty bonus, but it's confusing to understand

Reply:
I don't think I have enough knowledge xD